linker/validate.go

```diff
@@ -675,47 +680,49 @@ func (r *result) validateExtensionDeclarations(md *msgDescriptor, handler *repor
 				if err := handler.HandleErrorf(span, "extension declaration that is not marked reserved must have a full_name"); err != nil {
 					return err
 				}
-			}
-			var extensionFullName protoreflect.FullName
-			extensionNameSpan, _ := findExtensionRangeOptionSpan(r, md, i, extRange,
-				internal.ExtensionRangeOptionsDeclarationTag, int32(i), internal.ExtensionRangeOptionsDeclarationFullNameTag)
-			if !strings.HasPrefix(extDecl.GetFullName(), ".") {
-				if err := handler.HandleErrorf(extensionNameSpan, "extension declaration full name %q should start with a leading dot (.)", extDecl.GetFullName()); err != nil {
-					return err
-				}
-				extensionFullName = protoreflect.FullName(extDecl.GetFullName())
 			} else {
```

Comment:
This change and the one below move additional checks into an `else` block. Otherwise, we'd first complain that the name or type was missing, and then _also_ complain that the name or type (which was the empty string at this point) wasn't valid.